kaspresso/src/main/kotlin/com/kaspersky/kaspresso/runner/listener/KaspressoRunNotifier.kt

```diff
@@ -0,0 +1,70 @@
+package com.kaspersky.kaspresso.runner.listener
+
+import android.app.Instrumentation
+import org.junit.runner.Description
+import org.junit.runner.Result
+import org.junit.runner.notification.Failure
+import java.lang.IllegalStateException
+
+interface KaspressoRunNotifier : KaspressoRunListener {
```

Comment:
It is notified about the junit lifecycle as a listener and propagates it to the list of other listeners.